app/reducers/sessions.js

```diff
@@ -0,0 +1,165 @@
+import {
+    NEW_SESSION,
+    SWITCH_SESSION,
+    DELETE_SESSION,
+    UPDATE_CONNECTION,
+    UPDATE_CONFIGURATION,
+    UPDATE_IPC_STATE
+} from '../actions/sessions.js';
+import Immutable from 'immutable';
+import {EMPTY_SESSION} from '../constants/constants';
+
+// see end of file for `list` description
+const INITIAL_STATE = Immutable.Map({
+    list: Immutable.Map({}),
+    sessionSelected: 0,
+    startupIPC: Immutable.Map({})
+});
+
+export default function sessions(state = INITIAL_STATE, action) {
+
+    switch (action.type) {
+
+        /*
+         * TODO: make this action take a new random index
+         * instead of pushing to the list
+         */
+
+        case NEW_SESSION:
+            return state.mergeIn(['list'],
+                {
+                    [action.payload]: {
+                        configuration:
+                            Immutable.Map (EMPTY_SESSION.CONFIGURATION),
+                        connection:
+                            Immutable.Map(EMPTY_SESSION.CONNECTION),
+                        /*
+                         * include startup inter process communcation
+                         * into each session
+                         */
+                        ipc:
+                            state.get('startupIPC')
```

Comment:
This seems a lil odd to me. What is actually in `startupIPC` that we need to repeat across sessions?